youtube_dl/extractor/cda.py

```
@@ -0,0 +1,97 @@
+        ret = 0
+        for index, cipher in enumerate(word[::-1]):
+            ret += (base ** index) * self.dictionary[cipher]
+        return ret
+
+
+class CDAIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:(www|ebd)\.)?cda\.pl/(video|[0-9]+x[0-9]+)/(?P<id>[0-9a-z]+)'
+    _TEST = {
+        'url': 'http://www.cda.pl/video/5749950c',
+        'md5': '6f844bf51b15f31fae165365707ae970',
+        'info_dict': {
+            'id': '5749950c',
+            'ext': 'mp4',
+            'title': 'Oto dlaczego przed zakrętem należy zwolnić.',
+            'duration': 39,
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage("http://ebd.cda.pl/0x0/" + video_id, video_id)
+
+        if self._html_search_regex(r'(<a href="https:\/\/premium\.cda\.pl\/rejestracja" target="_blank" onclick="return premiumInfo\(\)")',
+                                   webpage, 'premium_check', default=None):
+            raise ExtractorError('This video is only available for premium users.', expected=True)
+
+        title = self._html_search_regex(r'<title>(.+?)<\/title>', webpage, 'title', fatal=False)
+        unpack = Unpacker()
+        durations = []
+
+        def _get_format(page):
+            m = re.search(r"}\('(?P<payload>.*)', *(?P<base>[0-9]+), *[0-9]+, *'(?P<symbol_table>.*)'\.split\('\|'\)",
+                          page, re.DOTALL)
+            unpacked = unpack(m.group('payload'), m.group('symbol_table').split('|'), int(m.group('base')))
+
+            try:
+                duration = self._search_regex(r"duration:\\'(.+?)\\'", unpacked, 'duration')
+                t = time.strptime(duration, '%H:%M:%S')
+                durations.append(datetime.timedelta(hours=t.tm_hour,
+                                                    minutes=t.tm_min,
+                                                    seconds=t.tm_sec).total_seconds())
+            except:
```

Never use bare except.